include/openenclave/internal/plugin.h

```diff
@@ -0,0 +1,467 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file plugin.h
+ *
+ * This file defines the programming interface for developing
+ * attester or verifier plugins for supporting alternative evidence formats.
+ *
+ */
+
+#ifndef _OE_ATTESTATION_PLUGIN_H
+#define _OE_ATTESTATION_PLUGIN_H
+
+#ifdef OE_BUILD_ENCLAVE
+// The attester related definitions are only available for the enclave.
+#include <openenclave/attestation/attester.h>
+#endif
+#include <openenclave/attestation/verifier.h>
+#include <openenclave/bits/evidence.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * Struct that defines the base structure of each attestation role plugin.
+ * Each plugin will have an UUID to indicate what evidence format
+ * is supported and have functions for registering/unregistering the plugin.
+ * Each plugin will also define the required function for their
+ * specific role (i.e., `get_evidence` for the attester and `verifiy_evidence`
+ * for the verifier).
+ */
+typedef struct _oe_attestation_role oe_attestation_role_t;
+struct _oe_attestation_role
+{
+    /**
+     * The UUID indicating what evidence format is supported.
+     */
+    oe_uuid_t format_id;
+
+    /**
+     * The function that gets executed when the plugin is registered.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the attestation role struct.
+     * @param[in] configuration_data An optional pointer to the configuration
+     * data.
+     * @param[in] configuration_data_size The size in bytes of
+     * configuration_data.
+     * @retval OE_OK on success.
+     * @retval An appropriate error code on failure.
+     */
+    oe_result_t (*on_register)(
+        oe_attestation_role_t* context,
+        const void* configuration_data,
+        size_t configuration_data_size);
+
+    /**
+     * The function that gets executed when the plugin is unregistered.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the attestation role struct.
+     * @retval OE_OK on success.
+     * @retval An appropriate error code on failure.
+     */
+    oe_result_t (*on_unregister)(oe_attestation_role_t* context);
+};
+
+#ifdef OE_BUILD_ENCLAVE
+
+/**
+ * The attester attestation role. The attester is reponsible for generating the
+ * attestation evidence and must implement the functions below.
+ */
+typedef struct _oe_attester oe_attester_t;
```

Comment:
>_oe_attester oe_attester_t [](start = 15, length = 26)

This is also defined in include/openenclave/plugin.h which I think you intend to remove.

Reply:
Yes, as described on docs/DesignDocs/Attestation_API_Proposal.md#proposed-oe-sdk-v010-public-api, the north and south APIs are separated into different header files.

Reply:
What I meant was - should "include/openenclave/attestation/plugin.h" be removed in this PR?